circuit/algorithms/src/poseidon/hash_to_scalar.rs

```diff
@@ -57,9 +57,11 @@ mod tests {
         let native = console::Poseidon::<<Circuit as Environment>::Network, RATE>::setup(DOMAIN)?;
         let poseidon = Poseidon::<Circuit, RATE>::constant(native.clone());
 
+        let rng = &mut TestRng::default();
```

Comment:
Same here, on the plus side moving the rng out will guard against this inefficiency in future.